CMake: Add support for all ARM SSG Mbed targets

hugueskamba opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->
CMake: Add support for all ARM SSG Mbed targets
<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [X] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

hugueskamba committed:
CMake: Add support for Arm's CM3DS_MPS2 target

hugueskamba committed:
CMake: Add support for Arm's MPS2 targets

hugueskamba committed:
CMake: Add mbed-psa library as a dependency to mbed-core if PSA is used

Include PSA source files only if a given Mbed target needs it.
This commit also consolidates CMakeLists.txt files within the platform
directory

hugueskamba committed:
CMake: Add support for ARM's MUSCA B1 target

Add CMakeLists.txt input source file for the target

hugueskamba committed:
CMake: Add support Arm Musca S1 target

Add CMakeLists.txt file for the target

hugueskamba committed:
Add support for all ARM SSG Mbed targets

rajkan01 reviewed (approved):
LGTM

0xc0170 commented on platform/FEATURE_EXPERIMENTAL_API/FEATURE_PSA/TARGET_MBED_PSA_SRV/CMakeLists.txt:
```diff
@@ -26,6 +34,37 @@ target_sources(mbed-psa
+        services/storage/common/psa_storage_common_impl.cpp
+
+        services/storage/its/pits_impl.cpp
+        services/storage/ts/psa_prot_internal_storage.cpp
```
an error comes from here, `its`, I'll fix and restart

hugueskamba replied:
Which error?

0xc0170 commented on platform/FEATURE_EXPERIMENTAL_API/FEATURE_PSA/CMakeLists.txt:
```diff
@@ -9,19 +9,13 @@ if("TFM" IN_LIST MBED_TARGET_LABELS)
-
-target_link_libraries(mbed-psa
-    INTERFACE
-        mbed-mbedtls
```
These 2 dependencies are missing in this pull request. The latest failure shows mbedtls must be included.

hugueskamba replied:
PSA was made part of `mbed-core` because some mbed-core code depended on it. Its use is gated by the presence of the `EXPERIMENTAL_API` and `PSA` labels.

0xc0170 reviewed (dismissed):
> CMake: Remove mbed-psa library as PSA is integral to mbed-core

Why psa is part of mbed-core? It used to be a feature, shouldn't it be? would it become part of baremetal?

hugueskamba committed:
CMake: Fix Cortex-M33-NS and Cortex-M33 ASM flags

The ASM flags were not supported by armclang, use the armclang alternatives.
Removed armlink --cpu flag as they generate a CMake configuration
time error. Furthermore, it is not necessary to specify the
processor or architecture as the linker auto-detects it from
the input object files.
See: https://developer.arm.com/documentation/101754/0615/armlink-Reference/armlink-Command-line-Options/--cpu-name--armlink-
